### PR TITLE
【Hackathon 6th No.17】为 Paddle 新增 sparse.mask_as API

### PR DESCRIPTION
**Pull request by megemini**
### PR Category

User Experience

### PR Types

Docs

### Description

[NO.17 为 Paddle 新增 sparse.mask_as API](https://github.com/PaddlePaddle/community/blob/master/hackathon/hackathon_6th/%E3%80%90Hackathon%206th%E3%80%91%E5%BC%80%E6%BA%90%E8%B4%A1%E7%8C%AE%E4%B8%AA%E4%BA%BA%E6%8C%91%E6%88%98%E8%B5%9B%E6%A1%86%E6%9E%B6%E5%BC%80%E5%8F%91%E4%BB%BB%E5%8A%A1%E5%90%88%E9%9B%86.md#no17-%E4%B8%BA-paddle-%E6%96%B0%E5%A2%9E-sparsemask_as-api)

关联 PR https://github.com/PaddlePaddle/Paddle/pull/64320

### CODE REVIEW

File: docs/api/paddle/sparse/mask_as_cn.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+-------------------------------
+
+.. py:function:: paddle.sparse.mask_as(x, mask, name=None)
+
+使用稀疏张量 `mask` 的索引过滤输入的稠密张量 `x`，并生成相应格式的稀疏张量。输入的 `x` 和 `mask` 必须具有相同的形状，且返回的稀疏张量具有与 `mask` 相同的索引，即使对应的索引中存在 `零` 值。
+
+参数
+:::::::::
+    - **x** (DenseTensor) - 输入的 DenseTensor。数据类型为 float32，float64，int32，int64，complex64，complex128，int8，int16，float16。
+    - **mask** (SparseTensor) - 输入的稀疏张量，是一个 SparseTensor，可以为 Coo 或 Csr 格式。当其为 SparseCsrTensor 时，应该是 2D 或 3D 的形式。
```
Comment (zhwesky2010) [inline docs/api/paddle/sparse/mask_as_cn.rst]: 这里直接写可为 `SparseCooTensor、SparseCsrTensor` 吧

File: docs/api/paddle/sparse/mask_as_cn.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    - **name** (str，可选) - 具体用法请参见 :ref:`api_guide_Name`，一般无需设置，默认值为 None。
+
+返回
+:::::::::
+SparseTensor: 其 Tensor 类型、dtype、shape 均与 `mask` 相同。
```
Comment (zhwesky2010) [inline docs/api/paddle/sparse/mask_as_cn.rst]: 其 稀疏格式

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+torch.Tensor.sparse_mask(mask)
+```
+
+### [paddle.sparse.mask_as](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/sparse/mask_as_cn.html)
+
+```python
+paddle.sparse.mask_as(x, mask, name=None)
+```
+
+PyTorch 作为 Tensor 的方法，Paddle 作为单独的函数调用，两者功能一致，参数用法一致，具体如下：
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md]: 这个看下其他的 `类方法转paddle方法` 的描述话术，例如 `torch.Tensor.hardshrink`

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+### 参数映射
+
+| PyTorch    | PaddlePaddle | 备注                                 |
+| ---------- | ------------ | ------------------------------------ |
+| -          | x            | 输入的 DenseTensor。                  |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md:21]: 这个看下其他的 `类方法转paddle方法` 的描述话术

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+## [ 仅 paddle 参数更多 ] torch.Tensor.sparse_mask
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md]: paddle的哪个参数更多？

**Comment by megemini** [comment]
> 这个看下其他的 `类方法转paddle方法` 的描述话术，例如 `torch.Tensor.hardshrink`
> 这个看下其他的 `类方法转paddle方法` 的描述话术
> paddle的哪个参数更多？

这几个问题统一回复一下：

torch 与 paddle 的这两个方法其实是一样的，不过，torch 是作为 Tensor 的方法调用，paddle 是单独的函数调用 ～

我找了找，还真没找到类似的函数，不太清楚具体该怎么写，所以写了：`仅 paddle 参数更多`，paddle 多了个输入 `x` ～

所以，到底该咋写呢？🤨

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+| PyTorch    | PaddlePaddle | 备注                                 |
+| ---------- | ------------ | ------------------------------------ |
+| -          | x            | 输入的 DenseTensor。                  |
+| mask       | mask         | 掩码逻辑的 mask，参数完全一致。           |
+
+### 转写示例
+
+```python
+# torch 版本可以直接访问方法
+# torch.sparse_mask(mask)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md]: 这个要写成类方法：`x.sparse_mask(mask)`

File: docs/api/paddle/sparse/mask_as_cn.rst [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+-------------------------------
+
+.. py:function:: paddle.sparse.mask_as(x, mask, name=None)
+
+使用稀疏张量 `mask` 的索引过滤输入的稠密张量 `x`，并生成相应格式的稀疏张量。输入的 `x` 和 `mask` 必须具有相同的形状，且返回的稀疏张量具有与 `mask` 相同的索引，即使对应的索引中存在 `零` 值。
+
+参数
+:::::::::
+    - **x** (DenseTensor) - 输入的 DenseTensor。数据类型为 float32，float64，int32，int64，complex64，complex128，int8，int16，float16。
+    - **mask** (SparseTensor) - 输入的稀疏张量，是一个 SparseTensor，可以为 SparseCooTensor、SparseCsrTensor。当其为 SparseCsrTensor 时，应该是 2D 或 3D 的形式。
```
Comment (zhwesky2010) [inline docs/api/paddle/sparse/mask_as_cn.rst]: `是一个 SparseTensor，` 这句话多余去掉

**Comment by megemini** [comment]
### Update 20240603

- 修改错误描述

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+torch.Tensor.sparse_mask(mask)
+```
+
+### [paddle.sparse.mask_as](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/sparse/mask_as_cn.html)
+
+```python
+paddle.sparse.mask_as(x, mask, name=None)
+```
+
+两者功能一致，但使用方式不一致，前者可以直接访问方法，后者需要调用方法，具体如下：
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md]: 两者功能一致，但调用方式不同，torch通过Tensor类方法调用，而paddle是直接调用函数，具体如下：

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+| PyTorch    | PaddlePaddle | 备注                                 |
+| ---------- | ------------ | ------------------------------------ |
+| -          | x            | 输入的 DenseTensor。                  |
+| mask       | mask         | 掩码逻辑的 mask，参数完全一致。           |
+
+### 转写示例
+
+```python
+# torch 版本可以直接访问方法
+# x = torch.tensor(123)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md]: 这里为啥要注释掉？

按这样写

```python
# torch 调用Tensor类方法
x.sparse_mask(mask)

# paddle 直接调用函数
paddle.sparse.mask_as(x, mask)
```

Comment (megemini) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.sparse_mask.md]: 参考的另一个 sparse 的文档 docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.is_sparse.md 🫠

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by sunzhongkai588 — APPROVED** [review]
LGTM
